bin/api-docs/gen-components-docs/index.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+const docgen = require( 'react-docgen-typescript' );
+const glob = require( 'glob' );
+const fs = require( 'fs' );
+const path = require( 'path' );
+
+/**
+ * Internal dependencies
+ */
+const { generateMarkdownDocs } = require( './markdown' );
+
+// For consistency, options should generally match the options used in Storybook.
+const OPTIONS = {
+	shouldExtractLiteralValuesFromEnum: true,
+	shouldRemoveUndefinedFromOptional: true,
+	propFilter: ( prop ) =>
+		prop.parent ? ! /node_modules/.test( prop.parent.fileName ) : true,
+	savePropValueAsString: true,
+};
+
+function getTypeDocsForComponent( {
+	manifestPath,
+	componentFilePath,
+	displayName,
+} ) {
+	const resolvedPath = path.resolve(
+		path.dirname( manifestPath ),
+		componentFilePath
+	);
+
+	return docgen
+		.parse( resolvedPath, OPTIONS )
```

Comment:
Could `docgen.parse()` potentially throw, or return an unexpected result?

Reply:
Not that I know of, but we'll see when we run more components through the system.